lib/uploadcare/service/upload_adapter.rb

```diff
@@ -23,25 +26,39 @@ def self.call(object, **options)
 
     def self.upload_file(object, **options)
       response = UploadClient.new.upload_many([object], **options)
+      handle_upload_errors(response)
       File.info(response.success.to_a.flatten[-1])
     end
 
     def self.upload_files(object, **options)
-      response = UploadClient.new.upload_many(object, **options)
+      response = handle_throttling { UploadClient.new.upload_many(object, **options) }
+      handle_upload_errors(response)
       Hashie::Mash.new(files: response.success.map { |pair| { original_filename: pair[0], uuid: pair[1] } })
     end
 
     def self.upload_big_file(object, **options)
       response = MultipartUploadClient.new.upload(object)
+      handle_upload_errors(response)
       File.new(response.success)
     end
 
+    def self.upload_from_url(object, **options)
+      response = UploadClient.new.upload_from_url(object, **options)
+      handle_upload_errors(response)
+      Uploader.new(response.success)
+    end
+
     def self.file?(object)
       object.respond_to?(:path) && ::File.exist?(object.path)
     end
 
     def self.big_file?(object)
       file?(object) && object.size >= MULTIPART_SIZE_THRESHOLD
     end
+
+    def self.handle_upload_errors(response)
```

Comment:
this doesn't handle errors, it "gets" them from the response

```suggestion
    def self.check_upload_errors(response)
```

also, shouldn't it be a part of UploadClient|MultipartUploadClient logic?